pandas/core/ops/__init__.py

```diff
@@ -680,13 +672,7 @@ def wrapper(self, other):
 
         res_values = comparison_op(lvalues, rvalues, op)
 
-        result = self._constructor(res_values, index=self.index)
-        result = finalizer(result)
-
-        # Set the result's name after finalizer is called because finalizer
-        #  would set it back to self.name
```

Comment:
Can you move this comment inside the construct_result function?

Reply:
good idea, will do